example reader.py

```diff
@@ -9,71 +10,122 @@
 
 class Hotel:
     def __init__(self, compact_hotel_info):
-        self.chain = compact_hotel_info.get("parent_chain_name")
-        self.name = compact_hotel_info.get("name")
-        self.id = compact_hotel_info.get("display_id")
-        
-        self.country = compact_hotel_info.get("country")
-        self.state_province = compact_hotel_info.get("state_province")
-        self.city = compact_hotel_info.get("city")
+        self.kw = compact_hotel_info
+    
+    @property
+    def chain(self):
+        return self.kw.get("parent_chain_name")
+
+    @property
+    def name(self):
+        return self.kw.get("name")
+
+    @property
+    def id(self):
+        return self.kw.get("display_id")
+
+    @property
+    def country(self):
+        return self.kw.get("country")
+
+    @property
+    def state_province(self):
+        return self.kw.get("state_province")
 
-        self.latitude = compact_hotel_info.get("latitude")
-        self.longitude = compact_hotel_info.get("longitude")
+    @property
+    def city(self):
+        return self.kw.get("city")
 
+    @property
+    def latitude(self):
+        return self.kw.get("latitude")
+
+    @property
+    def longitude(self):
+        return self.kw.get("longitude")
+    
     def display(self):
         print("chain: %s\nname: %s\nid: %s\ncity: %s\tstate_province: %s\tcountry: %s\nlatitude:\t%6.2f\nlongitude:\t%6.2f" % (self.chain, self.name, self.id, self.city, self.state_province, self.country, self.latitude, self.longitude))
 
 #class only intended for those with tag emit_auction_summary
 class RoomRequest:
     def __init__(self, **jsonObj):
-        raw_rooms = jsonObj.get("rooms")
-        if (raw_rooms != None):
-            self.rooms_requested = [RequestedRoom(room) for room in jsonObj.get("rooms")]
-        else:
-            self.rooms_requested = []
-        
-        self.potential_providers = jsonObj.get("potential_providers")
-
-        compact_hotel_info = jsonObj.get("compact_hotel_info")
-        if (compact_hotel_info != None):
-            self.hotel_info = Hotel(compact_hotel_info)
-        else:
-            self.hotel_info = None
-
-        raw_rates = jsonObj.get("raw_rates")
-        if (raw_rates != None):
-            self.rooms_found = [FoundRoom(rate) for rate in jsonObj.get("raw_rates")]
-        else:
-            self.rooms_found = []
+        self.kw = jsonObj
         
+        self.rooms_found = [FoundRoom(rate) for rate in jsonObj.get("raw_rates", [])]
+
+    @property
+    def rooms_requested(self):
+        return [RequestedRoom(room) for room in self.kw.get("rooms", [])]
+
+    @property
+    def potential_providers(self):
+        return self.kw.get("potential_providers")
+
+    @property
+    def hotel_info(self):
+        return Hotel(self.kw.get("compact_hotel_info", {}))
+
+    @property
+    def arrival_day(self):
+        return time.strptime(self.kw.get("arrival", ''), "%Y-%m-%d").tm_yday #day of the year [1, 366]

```

Comment:
I find that when I am writing code more than once, it is a good idea to make a function out of the reused code. Consider a method that takes a string formatted date and returns a time object. then you can reuse that method 
for arrival_day and departure_day